Student/Student/Assignments/AssignmentDetails/AssignmentDetailsPresenter.swift

```diff
@@ -89,6 +89,18 @@ class AssignmentDetailsPresenter: PageViewLoggerPresenterProtocol {
             }
         }
     }
+    var lockExplanation: String {
+        guard let assignment = assignment else { return "" }
+
+        switch assignment.lockStatus {
+        case .before:
+            return String.localizedStringWithFormat(NSLocalizedString("This assignment is locked until %@", comment: ""), (assignment.unlockAt ?? Date()).dateTimeString)
```

Comment:
Actually, those dates can't be nil because lockStatus uses those to determine the value of the enum, but I agree that it's a hidden connection so I added two guards here.